Feature/image upload

IsabellUrban opened the pull request.

IsabellUrban committed:
first commit

IsabellUrban committed:
install cloudinary and formidable

IsabellUrban committed:
add Upload Image input

IsabellUrban committed:
add function image uplad

IsabellUrban committed:
add api file

IsabellUrban committed:
rewrite uploadImage function and handleSubmit

IsabellUrban committed:
addImagePreview

IsabellUrban committed:
add schema for image

IsabellUrban committed:
add console.logs for debugging

IsabellUrban committed:
adjust api upload index.js

IsabellUrban committed:
add changes

IsabellUrban committed:
fix conditional handleSubmit

IsabellUrban committed:
add code suggestions

IsabellUrban committed:
update handleSubmit for creating flashcard without image

IsabellUrban committed:
reset code zu previous so that it works again

IsabellUrban committed:
add webkit-tap-highlight in styles

IsabellUrban committed:
add useState for image preview

IsabellUrban committed:
update handleImageClose

IsabellUrban committed:
fix image size on flashcard

IsabellUrban committed:
lift ste to app resolve issue

IsabellUrban committed:
remove uncommentcode

IsabellUrban committed:
add styles to enlarde image modal

IsabellUrban committed:
cleanup code

IsabellUrban committed:
fixed bug flashcard without image

IsabellUrban committed:
add smoot animation to enlarge Image

IsabellUrban committed:
remove console logs

IsabellUrban committed:
fix upload image with same url

IsabellUrban committed:
add changes

IsabellUrban committed:
solve conflicts

IsabellUrban committed:
fix bug

F-Kirchhoff commented on components/FormManual/FormManual.js:
```diff
@@ -105,6 +164,51 @@ export default function FormManual({
         )}
+
+        <StyledImageInputWrapper>
+          {(imageUploaded ||
+            (actionMode === "edit" &&
+              currentFlashcard?.imageUrl &&
+              showOriginalImage)) && (
```
you can refactor this into a variable:
```js
const showUploadPreview = imageUploaded ||
            (actionMode === "edit" &&
              currentFlashcard?.imageUrl &&
              showOriginalImage))
```
And then use this here instead:

```jsx
{showUploadPreview ? <Image Preview ...> : <Upload Button>}
```

IsabellUrban replied:
I have refactored it into an ImagePreview component.

F-Kirchhoff commented on components/FormManual/FormManual.js:
```diff
@@ -105,6 +164,51 @@ export default function FormManual({
+          )}
+          <>
+            <StyledImageInput
+              $hidden={
+                imageUploaded ||
+                (actionMode === "edit" &&
+                  currentFlashcard?.imageUrl &&
+                  showOriginalImage)
+              }
```
This is the exact same condition as the one above. Just use the `: <Upload Button>` strategy from the comment above instead of the $hidden attribute. By this, the button is only rendered if the statement is false.

F-Kirchhoff reviewed (changes requested):
Hey Brainstack Team,
this looks great, the toggling of the upload button and image preview might be a bit overly complicated, try to use the suggestion and see if it works.

IsabellUrban commented:
Hello Felix, thank you for your review. I have implemented the suggested changes. I created two components and then rendered them conditionally. Unfortunately, I found another error. When editing the flashcard, it was no longer possible to delete the image. That's why I adjusted the `handleSubmit` again with `const imageUrl = shouldRemoveImage
      ? null
      : url || currentFlashcard?.imageUrl || null;` 
Thanks for your help.

IsabellUrban committed:
implement suggestions and fix bug

F-Kirchhoff commented on components/FormManual/UploadButton.jsx:
```diff
@@ -0,0 +1,48 @@
+export default function UploadButton({ uploadImage }) {
+  return (
+    <>
+      <StyledImageInput>
+        <IconTextWrapper>
+          <Upload />
+          Upload Image
+        </IconTextWrapper>
+      </StyledImageInput>
+      <HiddenImageInput
+        name="image"
+        type="file"
+        accept="image/*"
+        id="image"
+        onChange={(event) => uploadImage(event.target.files[0])}
+      />
```
you seem to have two inputs for the file. can't you just use one, the styleImageInput and add the id, accpt, name type and onChange to that?

IsabellUrban replied:
Because I thought I needed the hidden one as the real upload button and since you can't style it I put the other styled one on top.

F-Kirchhoff reviewed (approved):
LGTM, there is an optional quesiton, but feel free to just merge.
